Testing/SystemTests/lib/systemtests/systemtesting.py

```diff
@@ -667,6 +671,46 @@ def start(self, script):
         os.remove(tmp_file.name)
         return results
 
+    def start_in_parent_thread(self, script):
+        """Run the given test code within the current Python process
+        Error handling adapted from: https://stackoverflow.com/questions/28836078/how-to-get-the-line-number-of-an-error-from-exec-or-execfile-in-python
+        Do not use in multithreading environment due to stdout context manager"""
+        exec_globals = dict()
+        exec_locals = dict()
+        exitcode = None
+        dual_stdout = DualStdOut()
```

Comment:
Can I just make sure what this is used for? We want to run the test and both print the output to the screen but also capture it to return to be store later?

Reply:
Yes that's right, it's an attempt the replicate the functionality from the Process object that gives you a dump of what that process (running a test) printed to stdout in order to collate the results